### PR TITLE
Support for testing enable/disable dr

### PR DESCRIPTION
**Pull request by nirs**
Split the channel and drpc from the subscription, so we can test deploying an app using OCM, and then enable and disable DR for the deployed application.

This will be used to add enable and disable DR flows to the ramen basic-test.

Tested with drenv regional-dr:
- [x] channel
- [x] subscription
- [x] busybox

Should be merged with https://github.com/RamenDR/ramen/pull/1080

### CODE REVIEW

File: subscription/subscription.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apps.open-cluster-management.io/v1
 kind: Subscription
 metadata:
   annotations:
-    apps.open-cluster-management.io/github-branch: main
```
Comment (ShyamsundarR) [inline subscription/subscription.yaml:6]: Same here, branch would eventually be `main` I assume

File: README.md [hunk 2]
```diff
@@ -43,6 +42,52 @@ OCM Stateful application samples, including Ramen resources.
...
+1. Change the PlacementRule to be reconciled by Ramen
+    - `kubectl patch placementrule busybox-placement -n busybox-sample --patch '{"spec.schedulerName": "ramen"}' --type merge`
+1. Deploy a DRPlacementControl resource for the OCM application on the
+   hub, for example:
+    - `kubectl apply -k dr/`
+    - The above creates a DRPlacementControl resource for the busybox
+    application in the `busybox-sample` namespace and can be viewed
+    using:
+        - `kubectl get drpc -n busybox-sample`
+    - At this point the application is managed by *Ramen*.
```
Comment (ShyamsundarR) [inline README.md:61]: (nit) At this point the application placement is managed by Ramen

**Review by ShyamsundarR — COMMENTED** [review]
Does not seem to affect downstream documentation. Changes are fine, except rolling back the testing related changes.

File: channel/channel.yaml [hunk 3]
```diff
@@ -6,4 +6,4 @@ metadata:
   namespace: ramen-samples
 spec:
   type: GitHub
-  pathname: https://github.com/RamenDR/ocm-ramen-samples.git
```
[the remark this answers is not in the document]
Comment (nirs) [reply channel/channel.yaml:9]: Yes I need to remove this commit, it was only for testing.

**Comment by nirs** [comment]
I'll remove the testing commit in the next version, marking as draft until it is removed.

File: README.md [hunk 4]
```diff
@@ -43,6 +42,51 @@ OCM Stateful application samples, including Ramen resources.
...
 
+## Undeploying the sample application
+
+To undeploy the application delete the subscription kustomization:
+    - `kubectl delete -k subscription/`
+
+## Enable DR for the OCM application
+
+1. Change the PlacementRule to be reconciled by Ramen
+    - `kubectl patch placementrule busybox-placement -n busybox-sample --patch '{"spec.schedulerName": "ramen"}' --type merge`
```
Comment (nirs) [inline README.md:53]: Wrong patch, should be:

    --patch '{"spec": {"schedulerName": "ramen"}}'

File: README.md [hunk 5]
```diff
@@ -43,6 +42,51 @@ OCM Stateful application samples, including Ramen resources.
...
+1. Ensure PlacementRule is pointing to the cluster where the workload is
+   currently placed to avoid data loss if OCM moves the application to
+   another cluster.
+   - The sample `busybox-placement` does not require any change.
+1. Delete the drpc resource for the OCM application on the hub, for example:
+    - `kubectl delete -k dr/`
+    - The above deletes the DRPlacementControl resource for the busybox
+    application, disabling replication and removing replicated data.
+1. Edit the `busybox-placement` resource and remove the `spec.schedulerName` property:
+    - `kubectl patch placementrule busybox-placement -n busybox-sample --patch '{"spec.schedulerName": null}' --type merge`
```
Comment (nirs) [inline README.md:74]: Wrong patch, should be:

    --patch '{"spec": {"schedulerName": null}}'

**Comment by nirs** [comment]
Code works, but the example kubectl command in the README is wrong.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**